skbio/stats/composition.py

```diff
@@ -608,10 +611,269 @@ def centralize(mat):
 
     """
     mat = closure(mat)
-    cen = ss.gmean(mat, axis=0)
+    cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.4.0-dev")
+def ancom(table, groups,
+          alpha=0.05,
+          tau=0.02,
+          theta=0.1,
+          multiple_comparisons_correction=None,
+          significance_test=scipy.stats.ttest_ind):
+    r""" Performs a differential abundance test using ANCOM
+
+    This is done by calculating pairwise log ratios between all features
+    and performing a signficance test to determine if there is a
+    significant difference in feature ratios with respect to the
+    variable of interest.
+
+    In an experiment with only two treatments, this test tests the
+    following hypothesis for feature :math:`i`
+
+    .. math::
+        H_{0i}:\mathbb{E}[\ln(u_i^{(1)})] = \mathbb{E}[\ln(u_i^{(2)})]
+
+    where :math:`u_i^{(1)}` is the mean abundance for feature
+    :math:`i` in the first group and :math:`u_i^{(2)}` is the
+    mean abundance for feature :math:`i` in the second group.
+
+    This method can be extended to an arbitrary number of classes
+    by passing in a different statistical function.
+
+    Parameters
+    ----------
+    table : pd.DataFrame or array_like
```

Comment:
Following up on this comment: @gregcaporaso and I chatted about the API and think it'll be safest and easiest to implement if the method only accepts a DataFrame (`table`) and Series (`grouping`). These structures should be [aligned](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.align.html) such that differences in sort order don't matter, but the same sample labels need to appear in both structures (error otherwise). This will provide a "strict" mode of operation and later on we can add other types of alignment (e.g., intersection).

We'd like to avoid `array_like` inputs because it can get tricky if one `array_like` input is provided but the other is a labeled structure. In that case, there can be odd alignment behavior because the unlabeled structure will be assigned labels 0..N, which could be misaligned if the other structure happens to have integer labels in a different order.

Thus, to avoid tricky cases like this, we think it'd be better to *only* support DataFrame and Series input and fail otherwise. We can always update the method later to accept `array_like` if there's demand for it. What do you think?